platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/internal/result/SingleNodeReconnectResultImpl.java

```diff
@@ -92,14 +94,26 @@ public List<SynchronizationCompletePayload> getSynchronizationCompletePayloads()
      * {@inheritDoc}
      */
     @Override
-    public void subscribe(@NonNull final ReconnectNotificationSubscriber subscriber) {
-        logResults.subscribe(logEntry -> {
-            final ReconnectNotification<?> notification = toReconnectNotification(logEntry);
-            if (notification != null) {
-                return subscriber.onNotification(notification);
-            }
-            return SubscriberAction.CONTINUE;
-        });
+    public void subscribe(@NonNull final ReconnectNotificationSubscriber newSubscriber) {
+        if (reconnectSubscribers.isEmpty()) {
+            reconnectSubscribers.add(newSubscriber);
+            logResults.subscribe(this::onLogEntry);
+        } else {
+            reconnectSubscribers.add(newSubscriber);
+        }
+    }
+
+    private SubscriberAction onLogEntry(@NonNull final StructuredLog logEntry) {
+        final ReconnectNotification<?> notification = toReconnectNotification(logEntry);
+        if (notification != null) {
+            reconnectSubscribers.forEach(subscriber -> {
+                if (subscriber.onNotification(notification) == SubscriberAction.UNSUBSCRIBE) {
+                    reconnectSubscribers.remove(subscriber);
```

Comment:
Have you tested this? I fear a `ConcurrentModificationException`. I suggest using `reconnectSubscribers.removeIf()`.